docs: tw796 add glossary entries for WASM module root and rollupEventBox (#3285)
OffchainLabs/arbitrum-docs: docs/partials/glossary/_wasm-module-root.mdx

---
title: WASM module root
key: wasm-module-root
titleforSort: WASM module root
---

> A 32-byte cryptographic hash that uniquely identifies a specific version of Arbitrum's [State Transition Function](/intro/glossary#state-transition-function) (STF), compiled to [WASM](/intro/glossary#wasm) for use in [fraud proofs](/intro/glossary#fraud-proof). It is computed as the Merkle root over the Keccak256 hashes of every module in the compiled prover image — the main STF replay binary plus runtime libraries such as `soft-float`, `host_io`, `user_host`, and `program_exec`.

Reviewer: ```suggestion
A 32-byte cryptographic hash that uniquely identifies a specific version of Arbitrum's [State Transition Function](/intro/glossary#state-transition-function) (STF), compiled to [WASM](/intro/glossary#wasm) for use in [fraud proofs](/intro/glossary#fraud-proof). It is computed as the Merkle root over the Keccak256 hashes of every module in the compiled prover image—the main STF replay binary plus runtime libraries such as `soft-float`, `host_io`, `user_host`, and `program_exec`.
```

The same WASM module root appears in two places:

> - **On-chain**, as the `wasmModuleRoot` field on the Rollup contract. It is set at deployment and updated by the chain owner via `setWasmModuleRoot`; every [Assertion](/intro/glossary#assertion) records the root used to compute it.

Reviewer: ```suggestion
- **Onchain**, as the `wasmModuleRoot` field on the Rollup contract. It is set at deployment and updated by the chain owner via `setWasmModuleRoot`; every [Assertion](/intro/glossary#assertion) records the root used to compute it.
```

> - **Off-chain**, by every [Validator](/intro/glossary#validator), as a compiled prover image under `target/machines/<root>/`. To validate or challenge an assertion, a validator must have the prover image matching the root recorded on **that** assertion — which may be the current on-chain root, or an older one if the assertion predates an [ArbOS](/intro/glossary#arbos) upgrade.

Reviewer: ```suggestion
- **Offchain**, by every [Validator](/intro/glossary#validator), as a compiled prover image under `target/machines/<root>/`. To validate or challenge an assertion, a validator must have the prover image matching the root recorded on **that** assertion—which may be the current onchain root, or an older one if the assertion predates an [ArbOS](/intro/glossary#arbos) upgrade.
```
